api/queries/blogs/getBlogDetails.ts

```diff
@@ -0,0 +1,19 @@
+import { useQuery } from '@tanstack/react-query';
+import QUERY_KEYS from '../../queryKeys';
+
+const fetchBlogDetails = async (id: string) => {
+  const response = await fetch(`${process.env.NEXT_PUBLIC_API_BASE_URL}/blog-posts/${id}`);
+  if (!response.ok) {
+    throw new Error(`HTTP error! Status: ${response.status}`);
+  }
+  return response.json();
```

Comment:
Here you should await the promise that resolves the json object and return the data.
The UI components shouldn't care about how we are fetching the data, what the data returns.